### ISSUE TITLE
how to deal with different endianess

### ISSUE THREAD

**Issue by sonoro1234** [gvx/bitser#26]
Hi,

If one machine calling dumps has different endianess from the machine calling loads there will be a problem.
How do you thing this should be managed?

**Comment by gvx** [comment]
To be honest, I had not yet considered this use-case! `ffi.abi('le')` can be used to identify the endianness of the current machine, and `bit.bswap` could be of some use, but I'm not sure that is going to be enough. I'll research this further.

**Comment by sonoro1234** [comment]
As ffi structs are binary saved as a whole it would be difficult to use bit.bswap I guess.

**Comment by sonoro1234** [comment]
Otherwise with a not binary serialization (as Ser for example) it wouldnt be an issue.

**Comment by gvx** [comment]
Yes, it doesn't seem to be possible to flip a switch for the endianness input or output of ffi functions. It may be necessary to use [`struct`](http://www.inf.puc-rio.br/~roberto/struct/) for that, but that still doesn't solve the problem for cdata... I think for now I'll mark it wontfix if you don't mind. I might add to the documentation that loading and dumping needs to be done on machines with the same endianness.
